sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/Command.scala

```diff
@@ -37,3 +37,9 @@ trait Command extends LogicalPlan {
 trait LeafCommand extends Command with LeafLike[LogicalPlan]
 trait UnaryCommand extends Command with UnaryLike[LogicalPlan]
 trait BinaryCommand extends Command with BinaryLike[LogicalPlan]
+
+trait CommandWithAnalyzedChildren extends Command {
+  def childrenToAnalyze: Seq[LogicalPlan] = Nil
+
+  override final def children: Seq[LogicalPlan] = childrenToAnalyze.filter(!_.analyzed)
```

Comment:
This is a bit risky. Some analyzer rules may want to transform the child even if it's analyzed.